src/main/java/org/opensearch/neuralsearch/search/query/HybridQueryPhaseSearcher.java

```diff
@@ -135,6 +140,24 @@ private void validateNestedBooleanQuery(final Query query, final int level) {
         }
     }
 
+    private void validateNestedDisJunctionQuery(final Query query, final int level) {
+        if (query instanceof HybridQuery) {
+            throw new IllegalArgumentException("hybrid query must be a top level query and cannot be wrapped into other queries");
+        }
+        if (level <= 0) {
+            // ideally we should throw an error here but this code is on the main search workflow path and that might block
+            // execution of some queries. Instead, we're silently exit and allow such query to execute and potentially produce incorrect
+            // results in case hybrid query is wrapped into such dis_max query
+            log.error("reached max nested query limit, cannot process dis_max query with that many nested clauses");
+            return;
+        }
+        if (query instanceof DisjunctionMaxQuery) {
```

Comment:
how it will work in case if one type of compound query is nested into another compound query type? E.g.
```
disjunction_max {
     bool {
            hybrid {
            }
      }
}
```

Reply:
This PR just covers when hybrid query is nested inside `disjunction_max`. We might need another issue to handle compound queries inside other ones